### DISCUSSION TITLE
Use Web API instead of TWS

### DISCUSSION THREAD

**Discussion by sebastianhaberey** [brndnmtthws/thetagang#511] [General]
Hi, I'm new to the IBKR APIs and I was looking for an example project to base my work on. Your project here is perfect, even in Python which I was also planning to use. I really love your idea of a terminal-based UI too, makes it a lot easier to run on SSH etc. So thanks for this!

One thing I noticed is that you're using the _ib_async / IBC / TWS_ toolchain for IBKR access. Is this because the Web API wasn't available when the project was started? Or does it still have too many limitations?

**Comment by brndnmtthws** [comment]
I haven't looked into it, but I'm guessing the web API doesn't have the same level of features/quality as the TWS/gateway API. Additionally, given the difficulty of handling auth/sessions with IBKR, I'd be worried that the web API will be even worse than TWS. IBKR really doesn't want you to automate too much, so they create a bunch of unnecessary speedbumps to prevent this (like requiring weird login flows instead of just allowing you to use a secret/token for auth).

**Comment by sebastianhaberey** [comment]
> IBKR really doesn't want you to automate too much

That would explain why the documentation is still so sparse. I couldn't find anything related to the OAuth 2.0 login process.
